core/services/relay/evm/functions/contract_transmitter.go

```diff
@@ -0,0 +1,232 @@
+			return fmt.Errorf("incorrect length of CoordinatorContract field: %d", len(requests[0].CoordinatorContract))
+		}
+		// NOTE: this is incorrect if batch contains requests destined for different contracts
+		// and there is currently no way areound it until we get rid of batching
+		destinationContract.SetBytes(requests[0].CoordinatorContract)
+	}
+
+	oc.lggr.Debugw("Transmitting report", "report", hex.EncodeToString(report), "rawReportCtx", rawReportCtx, "contractAddress", destinationContract, "txMeta", txMeta)
+
+	payload, err := oc.contractABI.Pack("transmit", rawReportCtx, []byte(report), rs, ss, vs)
+	if err != nil {
+		return errors.Wrap(err, "abi.Pack failed")
+	}
+
+	return errors.Wrap(oc.transmitter.CreateEthTransaction(ctx, destinationContract, payload, txMeta), "failed to send Eth transaction")
+}
+
+type contractReader interface {
+	CallContract(ctx context.Context, call ethereum.CallMsg, blockNumber *big.Int) ([]byte, error)
+}
+
+func parseTransmitted(log []byte) ([32]byte, uint32, error) {
+	var args abi.Arguments = []abi.Argument{
+		{
+			Name: "configDigest",
+			Type: utils.MustAbiType("bytes32", nil),
+		},
+		{
+			Name: "epoch",
+			Type: utils.MustAbiType("uint32", nil),
+		},
+	}
+	transmitted, err := args.Unpack(log)
+	if err != nil {
+		return [32]byte{}, 0, err
+	}
+	configDigest := *abi.ConvertType(transmitted[0], new([32]byte)).(*[32]byte)
+	epoch := *abi.ConvertType(transmitted[1], new(uint32)).(*uint32)
+	return configDigest, epoch, err
+}
+
+func callContract(ctx context.Context, addr common.Address, contractABI abi.ABI, method string, args []interface{}, caller contractReader) ([]interface{}, error) {
+	input, err := contractABI.Pack(method, args...)
+	if err != nil {
+		return nil, err
+	}
+	output, err := caller.CallContract(ctx, ethereum.CallMsg{To: &addr, Data: input}, nil)
+	if err != nil {
+		return nil, err
+	}
+	return contractABI.Unpack(method, output)
+}
+
+// LatestConfigDigestAndEpoch retrieves the latest config digest and epoch from the OCR2 contract.
+// It is plugin independent, in particular avoids use of the plugin specific generated evm wrappers
+// by using the evm client Call directly for functions/events that are part of OCR2Abstract.
+func (oc *contractTransmitter) LatestConfigDigestAndEpoch(ctx context.Context) (ocrtypes.ConfigDigest, uint32, error) {
+	latestConfigDigestAndEpoch, err := callContract(ctx, oc.contractAddress, oc.contractABI, "latestConfigDigestAndEpoch", nil, oc.contractReader)
+	if err != nil {
+		return ocrtypes.ConfigDigest{}, 0, err
+	}
+	// Panic on these conversions erroring, would mean a broken contract.
+	scanLogs := *abi.ConvertType(latestConfigDigestAndEpoch[0], new(bool)).(*bool)
+	configDigest := *abi.ConvertType(latestConfigDigestAndEpoch[1], new([32]byte)).(*[32]byte)
+	epoch := *abi.ConvertType(latestConfigDigestAndEpoch[2], new(uint32)).(*uint32)
+	if !scanLogs {
+		return configDigest, epoch, nil
+	}
+
+	// Otherwise, we have to scan for the logs.
```

Comment:
Do our contracts update the epoc in the OCR contract or do we need to scan the logs?  If we save the latest successfully transmitted epoch in the contracts, I wonder if that is easier to avoid needing to scan multiple contracts for `transmitted` logs.

Probably not a big deal either way, but something to consider. cc: @justinkaseman

Reply:
I believe we update in the contract. I'll think about it when I'm making a change here to support coordinators.